site/src/App/routes/gallery/Gallery.tsx

```diff
@@ -346,28 +357,35 @@ interface StageProps {
   title: string;
   jumpTo: JumpTo;
 }
-const Stage = ({ setName, jumpTo, title }: StageProps) => (
-  <Box data-braid-component-name={title}>
-    <Stack space="xxlarge">
-      <Box padding="xxlarge">
-        <Heading component="h2" level="1">
-          <span style={{ fontSize: '3em' }}>{title}</span>
-        </Heading>
-      </Box>
-      <Box>
-        {getRowsFor(setName).map((row, index) => (
-          <Columns space="none" key={index}>
-            {row.map((item) => (
-              <Column key={item.name} width="content">
-                <GalleryItem item={item} jumpTo={jumpTo} />
-              </Column>
-            ))}
-          </Columns>
-        ))}
-      </Box>
-    </Stack>
-  </Box>
-);
+const Stage = ({ setName, jumpTo, title }: StageProps) => {
+  const items = getRowsFor(setName);
+  const rowLength = items[0].length;
+
+  return (
+    <Box data-braid-component-name={title}>
+      <Stack space="xxlarge">
+        <Box padding="xxlarge">
+          <Heading component="h2" level="1">
+            <span style={{ fontSize: '3em' }}>{title}</span>
+          </Heading>
+        </Box>
+        <Box>
+          {items.map((row, index) => (
+            <Columns space="none" key={index}>
+              {range(rowLength).map((item) => (
+                <Column key={item} width={row[item] ? 'content' : undefined}>
```

Comment:
Won't this `key` potentially clash with the `key` 2 lines above? Might be worth prefixing with `column` and `row` or something.